File: solutions/leetcode/1752-check_if_array_is_sorted_and_rotated.java
class Solution {
    public boolean check(int[] nums) {
        int[] newNums = nums.clone();
        Arrays.sort(newNums);

        int n = nums.length;
        if (n == newNums.length) {
            for (int i = 0; i < n; i++) {
                boolean isRotation = true;
                for (int j = 0; j < n; j++) {
                    if (nums[j] != newNums[(i + j) % n]) {
                        isRotation = false;
                        break;
                    }
                }
                if (isRotation) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: solutions/leetcode/1752.check-if-array-is-sorted-and-rotated.cpp
/*
 * @lc app=leetcode id=1752 lang=cpp
 *
 * [1752] Check if Array Is Sorted and Rotated
 */

// @lc code=start
class Solution {
public:
    bool check(vector<int>& nums) {
        int count = 0;
        int n = nums.size();
        for (int i = 1; i < nums.size(); i++) {
            if(nums[i] < nums[i-1]){
                count++;
            }
        }
        if(nums[n-1] > nums[0]){
            count++;
        }
        return count <= 1;
    }
};
// @lc code=end

